docs/proposals/10-shoot-additional-container-runtimes.md

```diff
@@ -66,39 +77,35 @@ Each extension will need to address the following concern:
-                1. --container-runtime=remote
-                2. --container-runtime-endpoint=unix:///run/containerd/containerd.sock
-            2. Create a containerd configuration file: /etc/containerd/config.toml based on the current default configuration (containerd config default).
+            1. Override the ExecStart of ContainerD service to run with the configuration file in: /etc/containerd/config.toml and the correct ContainerD execution path in Ubuntu.
+            2. Create a cotainerruntime metadafile with: CONTAINER_RUNTIME_BIN_PATH=/usr/bin
         2. os-coreos - 
-            1. Add the following flags to the kubelet execution command:
-               1. --container-runtime=remote
-               2. --container-runtime-endpoint=unix:///var/run/docker/libcontainerd/docker-containerd.socket
-            2. Create a containerd configuration file: /etc/containerd/config.toml based on the default configuration.
-            3. Override the default containerd.service unit. Create a new unit in: /etc/systemd/system/containerd.service unit that will start the containerd service with a configuration file path: /etc/containerd/config.toml.
+            1. Override the ExecStart of Docker to contain the parameter: --host=fd:// --containerd=/run/containerd/containerd.sock and the correct ContainerD execution path in CoreOS.
+            2. Override the ExecStart of ContainerD service to run with the configuration file in: /etc/containerd/config.toml.
+            3. Create a cotainerruntime metadafile with: CONTAINER_RUNTIME_BIN_PATH=/var/bin
         3. os-suse-jeos - 
-           1. Add the following flags to the kubelet execution command:
-              1. --container-runtime=remote
-              2. --container-runtime-endpoint=unix:///var/run/docker/containerd/docker-containerd.sock 
-           2. Create a containerd configuration file: /etc/containerd/config.toml based on the default configuration.
-        
-    3. Docker-monitor daemon and conatinerd service. For CRI it is possible to send parameters to Kubelet to do the log rotation:
-       CRIContainerLogRotation=true. It will be done by the OS controller by default.
-       Containerd monitor will check that containerd process is running every 30 seconds.
-    4. Docker pull images in OSC should be changed to the Docker ctr equivalent command. 
-
-2. Existing running clusters created with Docker shim will not be modified until next scheduale rolling update or hibernation.
-   New clusters will be created with ContainerD by default.
+           1. Create ContainerD service unit with the relevant SuSe configurations.
+           2. Download and install ctr-cli.
+           3. Create a cotainerruntime metadafile with: CONTAINER_RUNTIME_BIN_PATH=/usr/sbin
+
+2. EnableContainerD property:
+   1. New clusters will be created with ContainerD by default (enableContainerD=true).
+   2. When .spec.hibernation.enabled goes from true to false enableContainerD will be set to true.
```

Comment:
See #1971, I think we can't enforce this yet without risking people to break.

Reply:
Please have a look at PR 1971 for my comment.